backend/test/testutil/test_utils.go

```diff
@@ -118,3 +116,18 @@ func ReplaceSDKInPipelineSpec(pipelineFilePath string, cacheDisabled bool, defau
 
 	return []byte(modifiedPipelineSpec)
 }
+
+func ReplaceBaseImageInPipelineSpec(pipelineFilePath string) []byte {
```

Comment:
some of these changes in the pr look like they don't exist upstream, can you specify if this is intended or not, and if we can have them upstream first, can we do that first?